nixos/modules/services/network-filesystems/s3fs-fuse.nix

```
@@ -0,0 +1,100 @@
+# Inspired by: https://discourse.nixos.org/t/how-to-setup-s3fs-mount/6283/2
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.s3fs-fuse;
+
+  mountModule = types.submodule {
+    options = {
+      mountPoint = mkOption {
+        type = types.str;
+        description = lib.mdDoc ''
+          The point where to mount the s3 filesystem. (second argument to s3fs)
+        '';
+        example = ''
+          "/mnt/s3"
+        '';
+      };
+
+      bucket = mkOption {
+        type = types.str;
+        description = lib.mdDoc ''
+          The name of the bucket you want to mount. (first argument to s3fs)
+        '';
+      };
+
+      options = mkOption {
+        type = types.listOf types.str;
+        description = lib.mdDoc ''
+          The options passed to the s3fs command
+        '';
+        example = ''
+          [
+            "passwd_file=/root/.passwd-s3fs"
+            "use_path_request_style"
+            "allow_other"
+            "url=https://ap-south-1.linodeobjects.com/" # Linode object storage
```

Not sure if we don't want to use a generic url here.

```suggestion
            "url=https://s3.example.com/"
```